src/libraries/System.Net.Http/tests/FunctionalTests/MultipartContentTest.cs

```diff
@@ -387,6 +388,128 @@ public void ReadAsStream_CreateContentReadStreamThrows()
+        [InlineData(false)]
+        public async Task ReadAsStream_CustomEncodingSelector_CustomEncodingIsUsed(bool async)
+        {
+            var mc = new MultipartContent("subtype", "fooBoundary");
+
+            var stringContent = new StringContent("bar1");
+            stringContent.Headers.Add("latin1", "\uD83D\uDE00");
+            mc.Add(stringContent);
+
+            var byteArrayContent = new ByteArrayContent(Encoding.ASCII.GetBytes("bar2"));
+            byteArrayContent.Headers.Add("utf8", "\uD83D\uDE00");
+            mc.Add(byteArrayContent);
+
+            byteArrayContent = new ByteArrayContent(Encoding.ASCII.GetBytes("bar3"));
+            byteArrayContent.Headers.Add("ascii", "\uD83D\uDE00");
+            mc.Add(byteArrayContent);
+
+            byteArrayContent = new ByteArrayContent(Encoding.ASCII.GetBytes("bar4"));
+            byteArrayContent.Headers.Add("default", "\uD83D\uDE00");
+            mc.Add(byteArrayContent);
+
+            mc.HeaderEncodingSelector = (name, _) => name switch
+            {
+                "latin1" => Encoding.Latin1,
+                "utf8" => Encoding.UTF8,
+                "ascii" => Encoding.ASCII,
+                _ => null
+            };
+
+            var ms = new MemoryStream();
+            if (async)
+            {
+                await (await mc.ReadAsStreamAsync()).CopyToAsync(ms);
+            }
+            else
+            {
+                mc.ReadAsStream().CopyTo(ms);
+            }
+
+            byte[] expected = Concat(
+                Encoding.Latin1.GetBytes("--fooBoundary\r\n"),
+                Encoding.Latin1.GetBytes("Content-Type: text/plain; charset=utf-8\r\n"),
+                Encoding.Latin1.GetBytes("latin1: "),
+                Encoding.Latin1.GetBytes("\uD83D\uDE00"),
+                Encoding.Latin1.GetBytes("\r\n\r\n"),
+                Encoding.Latin1.GetBytes("bar1"),
+                Encoding.Latin1.GetBytes("\r\n--fooBoundary\r\n"),
+                Encoding.Latin1.GetBytes("utf8: "),
+                Encoding.UTF8.GetBytes("\uD83D\uDE00"),
+                Encoding.Latin1.GetBytes("\r\n\r\n"),
+                Encoding.Latin1.GetBytes("bar2"),
+                Encoding.Latin1.GetBytes("\r\n--fooBoundary\r\n"),
+                Encoding.Latin1.GetBytes("ascii: "),
+                Encoding.ASCII.GetBytes("\uD83D\uDE00"),
+                Encoding.Latin1.GetBytes("\r\n\r\n"),
+                Encoding.Latin1.GetBytes("bar3"),
+                Encoding.Latin1.GetBytes("\r\n--fooBoundary\r\n"),
+                Encoding.Latin1.GetBytes("default: "),
+                Encoding.Latin1.GetBytes("\uD83D\uDE00"),
+                Encoding.Latin1.GetBytes("\r\n\r\n"),
+                Encoding.Latin1.GetBytes("bar4"),
+                Encoding.Latin1.GetBytes("\r\n--fooBoundary--\r\n"));
+
+            byte[] actual = ms.ToArray();
+
+            Assert.Equal(expected.Length, actual.Length);
+            for (int i = 0; i < expected.Length; i++)
```

Comment:
`Assert.Equal` is capable of [comparing array elements](https://github.com/xunit/assert.xunit/blob/726e48276895ecba39468990809664948113d876/Sdk/AssertEqualityComparer.cs#L177).
We might not need the loop here, unless there's some reason for it?

Reply:
> unless there's some reason for it?

Leftover from debugging, changed 👍